sw/device/tests/sim_dv/sysrst_ctrl_ulp_z3_wakeup_test.c

```diff
@@ -0,0 +1,186 @@
+  // This means 20 aon_clk ticks ~= 20 * 5 us = 100 us
+  kDebounceTimer = 20,
+};
+
+enum {
+  kTestPhaseInit = 0,
+  kTestPhaseDriveZero = 1,
+  kTestPhaseWaitNoWakeup = 2,
+  kTestPhaseGlitchLidOpen = 3,
+  kTestPhaseWaitWakeup = 4,
+  kTestPhaseDone = 5,
+};
+
+static const dif_pinmux_index_t kPeripheralInputs[] = {
+    kTopEarlgreyPinmuxPeripheralInSysrstCtrlAonPwrbIn,
+    kTopEarlgreyPinmuxPeripheralInSysrstCtrlAonAcPresent,
+    kTopEarlgreyPinmuxPeripheralInSysrstCtrlAonLidOpen,
+};
+
+static const dif_pinmux_index_t kInputPads[] = {
+    kTopEarlgreyPinmuxInselIor13,
+    kTopEarlgreyPinmuxInselIoc7,
+    kTopEarlgreyPinmuxInselIoc9,
+};
+
+static const dif_pinmux_index_t kPeripheralOutputs[] = {
+    kTopEarlgreyPinmuxOutselSysrstCtrlAonZ3Wakeup,
+};
+
+static const dif_pinmux_index_t kOutputPads[] = {
+    kTopEarlgreyPinmuxMioOutIob7,
+};
+
+/**
+ * Sets up the pinmux to assign input and output pads to the sysrst_ctrl
+ * peripheral as required.
+ */
+static void pinmux_setup(void) {
+  for (int i = 0; i < kNumMioInPads; ++i) {
+    CHECK_DIF_OK(
+        dif_pinmux_input_select(&pinmux, kPeripheralInputs[i], kInputPads[i]));
+  }
+
+  for (int i = 0; i < kNumMioOutPads; ++i) {
+    CHECK_DIF_OK(dif_pinmux_output_select(&pinmux, kOutputPads[i],
+                                          kPeripheralOutputs[i]));
+  }
+}
+
+/**
+ * Waits for `kTestPhase` variable to be changed by a backdoor overwrite
+ * from the testbench in chip_sw_<testname>_vseq.sv. This will indicate that
+ * the testbench is ready to proceed with the next phase of the test.
+ */
+static void wait_next_test_phase(void) {
+  // Set WFI status for testbench synchronization,
+  // no actual WFI instruction is issued.
+  test_status_set(kTestStatusInWfi);
+  test_status_set(kTestStatusInTest);
+  flash_ctrl_testutils_backdoor_wait_update(&flash, (uintptr_t)&kTestPhase,
+                                            kTestPhaseTimeoutUsec);
+}
+
+/**
+ * Configure *_debounce_ctl and then enable ULP wakeup.
+ */
+static void configure_wakeup(void) {
+  dif_sysrst_ctrl_ulp_wakeup_config_t wakeup_config;
+
+  // Keep toggle disabled when writing debounce configuration
+  wakeup_config.enabled = kDifToggleDisabled;
+  wakeup_config.ac_power_debounce_time_threshold = kDebounceTimer;
+  wakeup_config.lid_open_debounce_time_threshold = kDebounceTimer;
+  wakeup_config.power_button_debounce_time_threshold = kDebounceTimer;
+
+  CHECK_DIF_OK(
+      dif_sysrst_ctrl_ulp_wakeup_configure(&sysrst_ctrl, wakeup_config));
+
+  CHECK_DIF_OK(
+      dif_sysrst_ctrl_ulp_wakeup_set_enabled(&sysrst_ctrl, kDifToggleEnabled));
+}
+
+static void go_to_sleep(void) {
+  // Wakeup source is from sysrst_ctrl (source one).
+  LOG_INFO("Going to sleep.");
+  test_status_set(kTestStatusInWfi);
+  rstmgr_testutils_pre_reset(&rstmgr);
+  pwrmgr_testutils_enable_low_power(&pwrmgr, kDifPwrmgrWakeupRequestSourceOne,
+                                    /* pwrmgr_domain_config = */ 0);
```

Comment:
nit: To follow the style guide, remove spaces, i.e.: `/*pwrmgr_domain_config=*/ 0`